### PR TITLE
Add a description of the `AllowedMethods` module documentation

### PR DESCRIPTION
**Pull request by ydah**
This PR is fixes: https://github.com/rubocop/rubocop/pull/10829#issuecomment-1403082579

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [-] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [-] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [-] Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: lib/rubocop/cop/mixin/allowed_methods.rb [hunk 1]
```diff
@@ -3,7 +3,9 @@
 module RuboCop
   module Cop
     # This module encapsulates the ability to allow certain methods when
-    # parsing.
+    # parsing. Function to allow even code that would be an offense in this module.
```
Comment (bbatsov) [inline lib/rubocop/cop/mixin/allowed_methods.rb:6]: Function to allow even code that would be an offense in this module. - this sentence confuses me. 

Perhaps we'd be better off without it. You can also see the comment I left for the ticket that spawned this conversation.

Comment (ydah) [reply lib/rubocop/cop/mixin/allowed_methods.rb]: Thank you for your comments. I have improved the expression. How is it now?

File: lib/rubocop/cop/mixin/allowed_methods.rb [hunk 2]
```diff
@@ -3,7 +3,8 @@
 module RuboCop
   module Cop
     # This module encapsulates the ability to allow certain methods when
-    # parsing.
+    # parsing. Even if the code is in offense, if it contains methods
+    # that are allowed, it will deal with not be registered as an offense.
```
Comment (bbatsov) [inline lib/rubocop/cop/mixin/allowed_methods.rb:7]: "deal with" seems redundant here. 

You should probably also mentioned that this is functionality equivalent to the `IgnoredMethods` class that is deprecated and will be removed in RuboCop 2.0.

Comment (ydah) [reply lib/rubocop/cop/mixin/allowed_methods.rb]: Thank you so much. I updated this PR.

**Review by bbatsov — APPROVED** [review]

**Pull request merged**